apps/root/src/common/utils/earn/parsing.tsx

```diff
@@ -133,51 +138,61 @@ export const parseUserStrategies = ({
 
       let mappedHistory;
 
-      if ('history' in userStrategy) {
-        mappedHistory = userStrategy.history.map((action) => ({
-          ...action,
-          ...(action.action === EarnPositionActionType.WITHDREW
-            ? {
-                withdrawn: action.withdrawn.map((withdrawn) => ({
-                  ...withdrawn,
-                  token: sdkStrategyTokenToToken(
-                    withdrawn.token,
-                    `${strategy.farm.chainId}-${withdrawn.token.address}` as TokenListId,
-                    tokenList,
-                    strategy.farm.chainId
-                  ),
-                })),
-              }
-            : {}),
-        }));
-      }
-
-      return {
+      const baseEarnPosition: BaseEarnPosition | DetailedEarnPosition = {
         ...userStrategy,
         strategy,
+        history: [],
         balances: userStrategy.balances.map((balance) => ({
           ...balance,
           token: sdkStrategyTokenToToken(
             balance.token,
-            `${strategy.farm.chainId}-${balance.token.address}` as TokenListId,
+            `${strategy.network.chainId}-${balance.token.address}` as TokenListId,
             tokenList,
-            strategy.farm.chainId
+            strategy.network.chainId
           ),
         })),
-        history: mappedHistory,
         historicalBalances: userStrategy.historicalBalances.map((historicalBalance) => ({
           ...historicalBalance,
           balances: historicalBalance.balances.map((balance) => ({
             ...balance,
             token: sdkStrategyTokenToToken(
               balance.token,
-              `${strategy.farm.chainId}-${balance.token.address}` as TokenListId,
+              `${strategy.network.chainId}-${balance.token.address}` as TokenListId,
               tokenList,
-              strategy.farm.chainId
+              strategy.network.chainId
```

Comment:
Not following which chainId should be used here. Isn't it available in both farm and network? Don't mind if it's duplicated, but just to understand this change

Reply:
We should mostly try to use the network one, since its already filled with all the network details